### ISSUE TITLE
识别模型训练差异

### ISSUE THREAD

**Issue by lin-0015** [PaddlePaddle/PaddleOCR#4143]
请问下ocr版本release/2.1 release/2.2  release/2.3区别在哪呢？他们是分支？？直接就发布2.3？
为什么两段式算法中识别部分有些模型差别很大？
哪能去看代码提交记录吗，想看看当时是为什么那么修改

2.3版本训练backbone为MobileNetV3的db算法识别模型
一样的预训练模型 一样的配置参数 识别验证集准确率只有40%

2.1 可以达到 97%左右

**Comment by LDOUBLEV** [comment]
2.1到2.3分支的发布每次都更新了一些新的内容；

> 2.3版本训练backbone为MobileNetV3的db算法识别模型
> 一样的预训练模型 一样的配置参数 识别验证集准确率只有40%
> 2.1 可以达到 97%左右

你用的是哪个配置文件？

**Comment by lin-0015** [comment]
> 2.1到2.3分支的发布每次都更新了一些新的内容；
> 
> > 2.3版本训练backbone为MobileNetV3的db算法识别模型
> > 一样的预训练模型 一样的配置参数 识别验证集准确率只有40%
> > 2.1 可以达到 97%左右
> 
> 你用的是哪个配置文件？

rec_mv3_tps_bilstm_ctc

Global:
  use_gpu: True
  epoch_num: 3000
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: /paddle/PaddleOCR_v2.3/output/rec/mv3_tps_bilstm_ctc/
  save_epoch_step: 500
  # evaluation is run every 2000 iterations
  eval_batch_step: [0, 40]
  cal_metric_during_train: True
  pretrained_model: /paddle/PaddleOCR_v2.3/pretrain_models/rec_mv3_tps_bilstm_ctc_v2.0_train/best_accuracy
  # checkpoints: 
  checkpoints: /paddle/PaddleOCR_v2.3/output/rec/mv3_tps_bilstm_ctc/latest
  save_inference_dir: /paddle/PaddleOCR_v2.3/inference/rec_crnn/rec/
  use_visualdl: False
  infer_img: doc/imgs_words_en/word_10.png
  # for data or label process
  character_dict_path: ppocr/utils/number_dict.txt
  character_type: ch
  max_text_length: 25
  infer_mode: False
  use_space_char: False
  save_res_path: ./output/rec/predicts_r34_vd_tps_bilstm_ctc.txt
  debug : False
  distributed : True

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

Architecture:
  model_type: rec
  algorithm: STARNet
  Transform:
    name: TPS
    num_fiducial: 20
    loc_lr: 0.1
    model_name: small
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 96
  Head:
    name: CTCHead
    fc_decay: 0.0004

Loss:
  name: CTCLoss

PostProcess:
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: SimpleDataSet
    data_dir: /paddle/PaddleOCR_v2.3/train_data/
    label_file_list:
      - /paddle/PaddleOCR_v2.3/train_data/rec_gt.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 100]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: True
    batch_size_per_card: 256
    drop_last: True
    num_workers: 8
    use_shared_memory: False

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /paddle/PaddleOCR_v2.3/train_data/
    label_file_list:
      - /paddle/PaddleOCR_v2.3/train_data/eval_rec_gt.txt
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - CTCLabelEncode: # Class handling label
      - RecResizeImg:
          image_shape: [3, 32, 100]
      - KeepKeys:
          keep_keys: ['image', 'label', 'length'] # dataloader will return list in this order
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 256
    num_workers: 4
    use_shared_memory: False

**Comment by lin-0015** [comment]
> 2.1到2.3分支的发布每次都更新了一些新的内容；
> 
> > 2.3版本训练backbone为MobileNetV3的db算法识别模型
> > 一样的预训练模型 一样的配置参数 识别验证集准确率只有40%
> > 2.1 可以达到 97%左右
> 
> 你用的是哪个配置文件？

日志信息如下 再次确认配置文件和预训练模型的一致性
![image](https://user-images.githubusercontent.com/65577659/134616505-aa6ceaf1-079c-44bc-b640-77a45e97a43d.png)

![image](https://user-images.githubusercontent.com/65577659/134616545-fcc98a48-5789-4b79-8d19-3d017bb43cea.png)

**Comment by tink2123** [comment]
看两个log里的学习率不同且和官方配置并不一致，建议使用同一份配置文件训练，另外迭代轮次也不一致，请对比同一轮次的loss和acc。

**Comment by lin-0015** [comment]
> 看两个log里的学习率不同且和官方配置并不一致，建议使用同一份配置文件训练，另外迭代轮次也不一致，请对比同一轮次的loss和acc。
配置文件对照过了  日志信息前也有

![image](https://user-images.githubusercontent.com/65577659/134642894-f7b82fe6-4ebd-41ed-aa96-979ba27fcc79.png)


![image](https://user-images.githubusercontent.com/65577659/134643060-8a165108-5bd5-4ffc-b061-21b293b57faa.png)
![image](https://user-images.githubusercontent.com/65577659/134643170-468f9403-d168-49e3-9d7d-8ac7572699be.png)
![image](https://user-images.githubusercontent.com/65577659/134643338-c002928c-b664-4f8c-bb7a-c7e16a58e631.png)

**Comment by LDOUBLEV** [comment]
两个日志里的学习率不一样
![image](https://user-images.githubusercontent.com/26592129/134644023-979a39ee-e18e-453d-9ecb-19b46e33273e.png)

**Comment by lin-0015** [comment]
> 两个日志里的学习率不一样
> ![image](https://user-images.githubusercontent.com/26592129/134644023-979a39ee-e18e-453d-9ecb-19b46e33273e.png)

我之后改过 ,一致的也试了的

**Comment by lin-0015** [comment]
> 2.1到2.3分支的发布每次都更新了一些新的内容；
> 
> > 2.3版本训练backbone为MobileNetV3的db算法识别模型
> > 一样的预训练模型 一样的配置参数 识别验证集准确率只有40%
> > 2.1 可以达到 97%左右
> 
> 你用的是哪个配置文件？

![image](https://user-images.githubusercontent.com/65577659/134645786-0aaf6557-4299-4d98-a417-6c9b961a5f57.png)
![image](https://user-images.githubusercontent.com/65577659/134645931-f97da8b0-9b05-465e-b8e5-f2c8c1fe5c77.png)
![image](https://user-images.githubusercontent.com/65577659/134646634-14297c70-e714-4e5e-ba59-595cb75a3402.png)
2.3训练的比2.1快很多

**Comment by Gmgge** [comment]
ppocr2.3对应文章PP-OCRv2 Bag of Tricks for Ultra Lightweight OCR System，相对原先PP-OCR A Practical Ultra Lightweight OCR System做了一些改进，简单浏览下文章应该对该问题有所帮助。

**Comment by lin-0015** [comment]
> Bag of Tricks for Ultra Lightweight OCR System

好的 谢谢啦
